src/Platform/OpenGL/Shaders/god_ray_fragment.glsl

```diff
@@ -0,0 +1,57 @@
+#include "Library/directional_light.glsl"
+
+in vec2 TexCoord;
+out vec4 OutColor;
+
+uniform sampler2D albedoTex;
+uniform sampler2D normalTex;
+uniform sampler2D materialTex;
+uniform sampler2D depthTex;
+
+struct Camera
+{
+    vec3 position;
+    mat4 invViewProjMatrix;
+    mat4 viewProjMatrix;
+};
+
+uniform sampler2D cameraOutput;
+uniform Camera camera;
+
+uniform int lightCount;
+uniform DirLight lights[MaxDirLightCount];
+uniform sampler2D lightDepthMaps[MaxDirLightCount];
+
+uniform vec3 rayColor;
+
+void main()
+{
+    FragmentInfo fragment = getFragmentInfo(TexCoord, albedoTex, normalTex, materialTex, depthTex, camera.invViewProjMatrix);
+    vec3 camera2Frag = fragment.position-camera.position;
+    float fragDistance = length(camera2Frag);
+    vec3 fragDirection = normalize(camera2Frag);
+    vec3 currentColor = texture(cameraOutput, TexCoord).rgb;
+    
+    const float maxSteps = 255.0f;
```

Comment:
I would move this parameters to the C++ code and pass them as uniforms. Its better to have the ability to tweak them from the editor